arduino/cores/board.go

```diff
@@ -79,8 +79,17 @@ func (b *Board) buildConfigOptionsStructures() {
 
 	b.configOptions = properties.NewMap()
 	allConfigs := b.Properties.SubTree("menu")
-	for _, option := range allConfigs.FirstLevelKeys() {
-		b.configOptions.Set(option, b.PlatformRelease.Menus.Get(option))
+
+	// Used to show the config options in the same order as the menu, defined at the begging of platform.txt
+	if b.PlatformRelease.Menus != nil {
+		for _, menuOption := range b.PlatformRelease.Menus.FirstLevelKeys() {
+			for _, option := range allConfigs.FirstLevelKeys() {
```

Comment:
`allConfigs.FirstLevelKeys()` can be moved out of the loop? AFAICS is constant so we can compute it just once.

Reply:
BTW thinking about it a bit more...

Instead of `allConfigs.FirstLevelKeys()`, you can use `allConfigs.FirstLevelOff()` that returns a `map[string]*properties.Map`. Since you are testing the existence of `menuOption` in `allConfig` the check will be transformed into a map lookup instead of a loop.

You can also reuse the result of `allConfigs.FirstLevelOff()` later in the next loop, so basically there is no added complexity.

```go
    ...
	for option, optionProps := range allConfigs.FirstLevelOf() {   <---
		b.configOptionValues[option] = properties.NewMap()
		values := optionProps.FirstLevelKeys()
        ...
```

Reply:
I totally missed that function while implementing this :facepalm: Nice catch!